tools/flash-shelly.py

```diff
@@ -204,44 +243,91 @@ def get_current_version(self): # used when flashing between formware versions.
-               'SHGS-1' : 'ShellyGas',
-               'SHEM' : 'ShellyEM',
-               'SHEM-3' : 'Shelly3EM',
-               'SHSEN-1' : 'ShellySensor1',
-               'switch1' : 'Shelly1',
-               'switch1pm' : 'Shelly1PM',
-               'switch2' : 'Shelly2',
-               'switch25' : 'Shelly25',
-               'shelly-plug-s' : 'ShellyPlugS',
+    options = {'SHPLG-1' : ['ShellyPlug', 'shelly-plug'],
+               'SHPLG-S' : ['ShellyPlugS', 'shelly-plug-s'],
+               'SHPLG-U1' : ['ShellyPlugUS', 'shelly-plug-u1'],
+               'SHPLG2-1' : ['ShellyPlug', 'shelly-plug2'],
+               'SHSW-1' : ['Shelly1', 'switch1'],
+               'SHSW-PM' : ['Shelly1PM', 'switch1pm'],
+               'SHSW-L' : ['Shelly1L', 'switch1l'],
+               'SHSW-21' : ['Shelly2', 'switch'],
+               'SHSW-25' : ['Shelly25', 'switch25'],
+               'SHAIR-1' : ['ShellyAir', 'air'],
+               'SHSW-44' : ['Shelly4Pro', 'dinrelay4'],
+               'SHUNI-1' : ['ShellyUni', 'uni'],
+               'SHEM' : ['ShellyEM', 'shellyem'],
+               'SHEM-3' : ['Shelly3EM','shellyem3'],
+               'SHSEN-1' : ['ShellySensor', 'smart-sensor'],
+               'SHGS-1' : ['ShellyGas', 'gas-sensor'],
+               'SHSM-01' : ['ShellySmoke', 'smoke-sensor'],
+               'SHHT-1' : ['ShellyHT', 'ht-sensor'],
+               'SHWT-1' : ['ShellyFlood', 'water-sensor'],
+               'SHDW-1' : ['ShellyDoorWindow', 'doorwindow-sensor'],
+               'SHDW-2' : ['ShellyDoorWindow2', 'doorwindow-sensor2'],
+               'SHSPOT-1' : ['ShellySpot', 'spot'],
+               'SHCL-255' : ['ShellyColor', 'color'],
+               'SHBLB-1' : ['ShellyBulb', 'bulb'],
+               'SHCB-1' : ['ShellyColorBulb', 'color-bulb'],
+               'SHRGBW2' : ['ShellyRGBW2', 'rgbw2'],
+               'SHRGBWW-01' : ['ShellyRGBW2', 'rgbww'],
+               'SH2LED-1' : ['ShellyLED','led1'],
+               'SHDM-1' : ['ShellyDimmer', 'dimmer'],
+               'SHDM-2' : ['ShellyDimmer2', 'dimmer-l51'],
+               'SHDIMW-1' : ['ShellyDimmerW','dimmerw1'],
+               'SHVIN-1' : ['ShellyVintage', 'bulb6w'],
+               'SHBDUO-1' : ['ShellyDuo', 'bulbduo'],
+               'SHBTN-1' : ['ShellyButton1', 'wifi-button'],
+               'SHBTN-2' : ['ShellyButton2', 'wifi-button2'],
+               'SHIX3-1' : ['ShellyI3', 'ix3'],
     }
     return options.get(type, type)
 
+  def parse_local_file(self):
+    self.flash_label = "Local:"
+    if os.path.exists(self.local_file):
+      local_host = socket.gethostname()
+      s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+      s.connect((self.wifi_ip, 80))
+      local_ip = s.getsockname()[0]
+      logger.debug(f"Host IP: {local_ip}")
+      s.close()
+      with zipfile.ZipFile(self.local_file, "r") as zfile:
+        for name in zfile.namelist():
+          if re.search(r'\.json$', name) is not None:
```

Comment:
you need `manifest.json`, not just any .json

Reply:
oops ye, https://github.com/mongoose-os-apps/shelly-homekit/pull/492/commits/691edaadb9c2112c54d48669c97308c401e374cd fixes that.